app/controllers/api/pages_controller.rb

```diff
@@ -14,13 +16,54 @@ def update
   end
 
   def share_rows
+    get_page
     render json: (@page.shares.map do |s|
       {html: render_to_string(partial: "share/#{s.name}s/summary_row", locals: {share: s, page: @page})}
     end)
   end
 
+  def show
+    if !show_single_page
```

Comment:
I'm not really big on the whole human readable syntax as a whole so I'm not a big fan of `unless`. I can change it if you like but it makes no real difference, tbh.